src/conf.c

```diff
@@ -249,7 +250,7 @@ struct {
                  "(" "(none)" WS STR ")|" \
                  "(" "(http|socks4|socks5)" WS \
                      "(" USERNAME /*username*/ ":" PASSWORD /*password*/ "@" ")?"
-                     "(" IP "|" ALNUM ")"
+                     "(" IP "|" "\\[(" IPV6 ")\\]" "|" ALNUM ")"
```

Comment:
in other contexts where IPv6 is accepted, it is accepted without brackets around them; we should be consistent

Reply:
i realized that the bracket syntax stems from the fact that the colon character used to separate ip from port can't properly be distinguished from the ones used inside the ipv6 literal. can you research whether upstream directive is the only spot where an ip:port pair is used ?

Reply:
IPV6 addresses are accepted by  listen, allow, deny, bind and upstream. upstream is the only direcdtive, that also allows to specify a port number.

The syntax using square brackes vor IPv6 adresses is taken from the URL syntax for numerical IPv6 adresses. I must admit, that it looks ugly to me, but I had no better idea. :-(